synapse/rest/client/v2_alpha/account.py

```diff
@@ -107,21 +116,35 @@ async def on_POST(self, request):
 
             raise SynapseError(400, "Email not found", Codes.THREEPID_NOT_FOUND)
 
+        # Get the email addresses of the user from database
+        # The user can own more than one address. Lookup for the right address.
+        # The email will be sent to the stored address.
+        # It should prevent potential account hijack via password reset form,
+        # if some compare algorithm are not exactly.
```

Comment:
this doesn't make much sense...

```suggestion
        # This avoids a potential account hijack by requesting a password reset to
        # an email address which is controlled by the attacker but which, after 
        #  canonicalisation, matches the one in our database.
```